test/type/lang.tst.ts

```diff
@@ -4,11 +4,10 @@ import { expect, test } from "tstyche";
 import { Awaitable, ItemOfIterable, noop } from "../../src/lang.ts";
 
 test("Awaitable", () => {
-	expect(11).type.toMatch<Awaitable<number>>();
-	expect(Promise.resolve(11)).type.toMatch<Awaitable<number>>();
+	expect<Awaitable<number>>().type.toEqual<number | PromiseLike<number>>();
 
-	expect<void>().type.toMatch<Awaitable<void>>();
-	expect<Promise<void>>().type.toMatch<Awaitable<void>>();
+	expect<Awaitable<number>>().type.toBeAssignable(11);
+	expect<Awaitable<number>>().type.toBeAssignable(Promise.resolve(11));
```

Comment:
Previously `expectAssignable()` was used here. I was wondering why you didn’t swap it with `.toBeAssignable()`? 

One user was suggesting that it is not clear what `.toBeAssignable()` does and was suggesting to rename it to `.toBeAssignableWith()`. Perhaps this is the missing bit?

Reply:
Because my thought here is "Can this type of value be used as `Awaitable`?". As you can see the word value comes before `Awaitable`, so I'm representing this in the code.

Maybe `expect(11).type.toBeAssignableTo<Awaitable<number>>()` is a good choice.

Reply:
Yes, we are used to see "assignable to". But isn’t it odd have type which is under the test on right side? My thinking was that it should always stay on left side of assertion.

In the other hand, the following

```ts
expect<Awaitable<number>>().type.toEqual<number | PromiseLike<number>>();
```

tests that `Awaitable<number>` will be assignable to `number` or `PromiseLike<number>`. It is a union type and this is how it will be handled by TypeScript.

Reply:
To keep type which is under test on left side always, `.toBeAssignable()` is implemented just opposite from "assignable to". TypeScript's spec defines this relation as "assignable from", but `.toBeAssignableFrom()` sounds somewhat very unusual. That’s why `.toBeAssignableWith()` was considered.

> "Can this type of value be used as Awaitable?"

This test is here:

```ts
expect<Awaitable<number>>().type.toBeAssignable(11);
expect<Awaitable<number>>().type.toBeAssignable(Promise.resolve(11));
```

Somewhat similar to (same order of elements, but not exact behaviour):

```ts
const x: Awaitable<number> = 11 ;
const x: Awaitable<number> = Promise.resolve(11);

```

Reply:
Hm.. In a way `.toMatch()` does check "Can this type of value be used as Awaitable?", but it isn't 100% clear that assignability is checked. I mean, reading this code as a strange I couldn't get that you are checking assignability.

Perhaps having both `.toBeAssignableFrom()` and `.toBeAssignableTo()` is the solution? Altho can be puzzling.

In any case, this is your repo and you are the boss here. I just wanted to hear some feedback.